baserun/baserun.py

```diff
@@ -20,245 +39,324 @@ class BaserunEvaluationFailedException(Exception):
 
 class Baserun:
     _initialized = False
-    _trace_id = None
-    _traces = []
-
-    _buffer = []
-    _buffer_lock = threading.Lock()
-
-    _evals = []
-    _evals_lock = threading.Lock()
 
     _api_base_url = None
     _api_key = None
 
+    _instrumentors: list[BaseInstrumentor] = None
+
+    submission_service: SubmissionServiceStub = None
+
+    current_test_suite: TestSuite = None
+
     evals = Evals
 
     @staticmethod
     def init(api_base_url: str = "https://baserun.ai/api/v1") -> None:
-        api_key = os.environ.get('BASERUN_API_KEY')
+        api_key = os.environ.get("BASERUN_API_KEY")
         if not api_key:
-            raise ValueError("Baserun API key is missing. Ensure the BASERUN_API_KEY environment variable is set.")
+            raise ValueError(
+                "Baserun API key is missing. Ensure the BASERUN_API_KEY environment variable is set."
+            )
+
+        grpc_base = os.environ.get("BASERUN_GRPC_URI", "grpc.baserun.ai:50051")
 
         if Baserun._initialized:
-            warnings.warn("Baserun has already been initialized. Additional calls to init will be ignored.")
+            warnings.warn(
+                "Baserun has already been initialized. Additional calls to init will be ignored."
+            )
             return
 
         Baserun._api_base_url = api_base_url
         Baserun._api_key = api_key
         Baserun._initialized = True
-        Baserun.evals.init(Baserun._append_to_evals)
 
-        OpenAIWrapper.init(Baserun._handle_auto_llm)
-        AnthropicWrapper.init(Baserun._handle_auto_llm)
+        if key_chain := os.environ.get("SSL_KEY_CHAIN"):
+            ssl_creds = grpc.ssl_channel_credentials(
+                root_certificates=bytes(key_chain, "utf-8")
+            )
+        else:
+            ssl_creds = grpc.ssl_channel_credentials()
+
+        call_credentials = grpc.access_token_call_credentials(api_key)
+        channel_credentials = grpc.composite_channel_credentials(
+            ssl_creds, call_credentials
+        )
+        grpc_channel = grpc.secure_channel(grpc_base, channel_credentials)
+        Baserun.submission_service = SubmissionServiceStub(grpc_channel)
+
+        tracer_provider = trace.get_tracer_provider()
+        tracer = tracer_provider.get_tracer("baserun")
```

Comment:
Should this line move to after the if block after we've replaced the tracerprovider?

Reply:
The only reason we're fetching the tracer is to see if we have an active span processor. I added a comment to clarify.